compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.h

```diff
@@ -7,6 +7,8 @@
 #ifndef IREE_AMD_AIE_TRANSFORMS_KERNELDISPATCH_H_
 #define IREE_AMD_AIE_TRANSFORMS_KERNELDISPATCH_H_
 
+#include "iree-amd-aie/aie_runtime/AMDAIEEnums.h"
+#include "iree-amd-aie/aie_runtime/iree_aie_runtime.h"
```

Comment:
I don't think that this include is needed in the header? It should be put in the cpp file.

Reply:
No, these are required here because of the following reasons :-

1. I use `AMDAIEDevice` [here](https://github.com/nod-ai/iree-amd-aie/blob/fc0480187f9dabb2cbf077a3ed643a8fd5cf6fca/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.h#L43) within the same file.

2. [PassDetail.h](https://github.com/nod-ai/iree-amd-aie/blob/e25d3484be8f53b89aff1d51a9180c2c3768c14d/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/PassDetail.h#L10) includes KernelDispatch.h -> and since this is how the [`.td` derives the types/attr too](https://github.com/nod-ai/iree-amd-aie/blob/e25d3484be8f53b89aff1d51a9180c2c3768c14d/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/PassDetail.h#L86).

Reply:
AMDAIEDevice is defined in `AMDAIEEnums.h.inc`, so I don't think you need `iree_aie_runtime.h`

Reply:
Sure, I've moved `iree_aie_runtime.h` from this file and included it in `KernelDispatch.cpp` and `InsertLoopsForVectorization.cpp`.